devdoc/doublylinkedlist_requirements.md

```diff
@@ -89,10 +89,10 @@ DList_InsertHeadList shall assume listHead and listEntry non-NULL and pointing t
 
 ### DList_AppendTailList
 ```c
-void DList_AppendTailList(PDLIST_ENTRY listHead, PDLIST_ENTRY ListToAppend);
+void DList_AppendTailList(PDLIST_ENTRY listHead, PDLIST_ENTRY listToBeAppendedHead);
 ```
 
-**SRS_DLIST_06_007: [** DList_AppendTailList shall place the list defined by listToAppend at the end of the list defined by the listHead parameter. **]**
+**SRS_DLIST_06_007: [** DList_AppendTailList shall place the list defined by `listToBeAppendedHead` at the end of the list defined by the `listHead` parameter. **]**
```

Comment:
>_DLIST_06_007: [** DList_AppendTailList shall place the list defined by `listToBeAppendedHead` at the end of the list defined by the `listHead` parameter. **]** [](http://example.com/codeflow?start=5&length=160)

Hmmm, the whole headless list vs a list with a head made me wonder again.
I remember that Tony took these from NT.
And lo and behold, here is the docs for one instance of it:

[https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-appendtaillist](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-appendtaillist) 

The docs call out explicitly that this is a headless list.
I think I am very curious now how this worked fine for generations and it does not work for us.